src/components/ReportsPage/HeatmapPage/Tables/SingleScorecardHeatmapTable.tsx

```diff
@@ -55,9 +62,26 @@ export const SingleScorecardHeatmapTable = ({
     [levelsDriven, ladder, scores],
   );
 
+  const { value: teamHierarchies, loading: loadingTeamHierarchies } = useCortexApi(api => api.getTeamHierarchies());
+  const { value: domainHierarchies, loading: loadingDomainHierarchies } = useCortexApi(api => api.getDomainHierarchies());
+
   const data = useMemo(() => {
-    return getScorecardServiceScoresByGroupByOption(scores, groupBy);
-  }, [scores, groupBy]);
+    const groupedData = getScorecardServiceScoresByGroupByOption(scores, groupBy, domainTagByEntityId);
+
+    if (useHierarchy) {
+      if (groupBy === GroupByOption.TEAM && teamHierarchies) {
+        return groupScoresByHierarchies(groupedData, teamHierarchies.orderedParents);
+      } else if (groupBy === GroupByOption.DOMAIN && domainHierarchies) {
+        return groupScoresByHierarchies(groupedData, domainHierarchies.orderedTree);
+      }
+    }
+
+    return groupedData;
+  }, [scores, groupBy, useHierarchy, teamHierarchies, domainHierarchies, domainTagByEntityId]);
+
+  if (useHierarchy && ((groupBy === GroupByOption.TEAM && loadingTeamHierarchies) || (groupBy === GroupByOption.DOMAIN && loadingDomainHierarchies))) {
+    return <Progress />
```

Comment:
Oof Prettier doesn't run in this repo and clearly ESLint is not enforcing consistent semicolon usage 😓